jenkins.sh

```diff
@@ -130,16 +132,19 @@ else
   exit 1
 fi
 
-R -e ".libPaths('../RLIB');\
-  setwd(sprintf('%s/tests', getwd()));\
-  source('testthat.R')"
+echo ".libPaths('../RLIB');" > runTestthat.R
+echo "setwd(sprintf('%s/tests', getwd()));" >> runTestthat.R
+echo "source('testthat.R')" >> runTestthat.R
+R --vanilla < runTestthat.R
+rm runTestthat.R
   
 # test that load works after detach
-R -e ".libPaths('../RLIB');\
-  library(PythonEmbedInR);\
-  detach("package:PythonEmbedInR", unload=TRUE);\
-  library(PythonEmbedInR)"
-   
+echo ".libPaths('../RLIB');" > testDetach.R
```

Comment:
I think it IS OK to combine the two R sessions into one.  It's important to keep the library-detach-library sequence within the same R session to test that re-attachment works.